indigo_app/views/places.py

```diff
@@ -100,8 +100,133 @@ def get_context_data(self, **kwargs):
+        context['open_tasks_by_label'] = self.calculate_open_tasks()['labels_chart']
+        context['subtypes'] = self.get_works_by_subtype(works)
+
+        # place activity
+        since = now() - timedelta(days=14)
+        metrics = DailyPlaceMetrics.objects \
+            .filter(country=self.country, locality=self.locality, date__gte=since) \
+            .order_by('date') \
+            .all()
+
+        context['activity_history'] = json.dumps([
+            [m.date.isoformat(), m.n_activities]
+            for m in metrics
+        ])
+
+        # stubs overview
+        context['stubs_count'] = works.filter(stub=True).count()
+        context['non_stubs_count'] = works.filter(stub=False).count()
+        context['stubs_percentage'] = int((context['stubs_count'] / works.count()) * 100)
+        context['non_stubs_percentage'] = 100 - context['stubs_percentage']
+
+        # primary works overview
+        context['primary_works_count'] = works.filter(parent_work__isnull=True).count()
+        context['subsidiary_works_count'] = works.filter(parent_work__isnull=False).count()
+        context['primary_works_percentage'] = int((context['primary_works_count'] / works.count()) * 100)
+        context['subsidiary_works_percentage'] = 100 - context['primary_works_percentage']
+
+        # Completeness
+        metrics = list(DailyWorkMetrics.objects
+            .filter(place_code=self.place.place_code)
+            .filter(date__gte=since)
+            .order_by('date')
+            .all())
+
+        if metrics:
+            context['latest_stat'] = metrics[-1]
+
+        # breadth completeness history
+        context['completeness_history'] = json.dumps([
+            [m.date.isoformat(), m.p_breadth_complete]
+            for m in metrics])
+
+        return context
+
+    def get_recently_updated_works(self):
+        # TODO: this should also factor in documents that have been updated
+        return Work.objects \
+            .filter(country=self.country, locality=self.locality) \
+            .order_by('-updated_at')[:5]
+
+    def get_recently_created_works(self):
+        return Work.objects \
+                   .filter(country=self.country, locality=self.locality) \
+                   .order_by('-created_at')[:5]
+
+    def get_works_by_subtype(self, works):
+        def subtype_name(abbr):
+            if not abbr:
+                return 'Act'
+            st = Subtype.for_abbreviation(abbr)
+            return st.name if st else abbr
+        pairs = list(Counter([subtype_name(w.subtype) for w in works]).items())
+        pairs = [list(p) for p in pairs]
+        pairs.sort(key=lambda p: p[1], reverse=True)
+
+        total = sum([x[1] for x in pairs])
+        for p in pairs:
+            p.append(int((p[1] / (total or 1)) * 100))        
+
+        return pairs
+
+    def calculate_open_tasks(self):
+        tasks = Task.objects.filter(country=self.country, locality=self.locality) \
```

Comment:
Use `Task.objects.unclosed().filter(...)` and the drop the state filters. The Task object manager will do the work.